### PR TITLE
[System] Adds Windows XP support and moves the registering of the GUID's to the main thread

### PR DESCRIPTION
**Pull request by kdschlosser**
Windows XP does not support the use of RegisterPowerNotification and UnregisterPowerNotificatrion it always sends the notification. I want to thank Diz for providing the solution to fix this. Tho I did modify it so that in the future any new Windows versions will force a review of the code. I did this because of how Microsoft likes to change things about. We do want to make sure we have a working plugin. So in the event that there is an unsupported GUID in a new OS it will not register the GUID's because all of the OS checks will return False. 

The Registering of the GUID's is much happier being done in the main thread. even tho it does appear to work if done in another thread, it does not function properly. I discovered this issue when working with another part of EG that also uses win32com and when i had an error while working with that it would also cause the PowerBroadcastNotifier to throw a traceback. and only the PowerBroadcastNotifier would even tho the use of win32com is use throughout EG. and one of the traceback has some information in it about not being run in the main thread. 

And now since it is done in the main thread there are a couple of new events on startup as well. this is a good indicator it is functioning properly. And I believe this is the fix for #181. Not setting those GUID's in the main thread really mucks up odd parts of pywin32. I feel this is something that we may need to monkey patch pywin32 to have it throw an exception if it is not being done in the main thread as to avoid any future issues, IE: my forgetfulness or even a plugin dev. I suggest this because there is no indication there is a problem just random weird things happening.

### CODE REVIEW

**Comment by kdschlosser** [comment]
Do not merge this yet when #208 or a variation of gets merged I will update this to use the eg.WindowsVersion

**Comment by kdschlosser** [comment]
OK I have made the changes to support the new eg.WindowsVersion. so everything should be good to go on this...  I have been using the changes to how the messages are registered for some while without any errors.. and I know that @per1234 has used this as well... I have not heard a report back.. so I am going to take it as no news is good news.

File: plugins/System/PowerBroadcastNotifier.py [hunk 1]
```diff
@@ -62,10 +62,15 @@
 AWY_EXITING = 0x0
 AWY_ENTERING = 0x1
 
-if WIN_7:
-    monGUID = '{02731015-4510-4526-99e6-e5a17ebd1aea}'
+
+if eg.WindowsVersion >= 8:
```
Comment (topic2k) [inline plugins/System/PowerBroadcastNotifier.py]: needs to be: `if eg.WindowsVersion() >= 8:`

File: plugins/System/PowerBroadcastNotifier.py [hunk 2]
```diff
@@ -172,38 +188,40 @@ class POWERSCHEME_PERSONALITY(ctypes.Structure):
...
     def __init__(self, plugin):
         self.plugin = plugin
+        self.monitorNotify = None
+        self.awayNotify = None
+        self.sourceNotify = None
+        self.batteryNotify = None
+        self.savingNotify = None
+        self.schemeNotify = None
 
+        if eg.WindowsVersion >= 'Vista':
```
Comment (topic2k) [inline plugins/System/PowerBroadcastNotifier.py]: needs to be `if eg.WindowsVersion() >= 'Vista':`

File: plugins/System/PowerBroadcastNotifier.py [hunk 3]
```diff
@@ -172,38 +188,40 @@ class POWERSCHEME_PERSONALITY(ctypes.Structure):
...
-        )
-
     def Close(self):
-        Unregister(self.monitorNotify)
-        Unregister(self.awayNotify)
-        Unregister(self.sourceNotify)
-        Unregister(self.batteryNotify)
-        Unregister(self.savingNotify)
-        Unregister(self.schemeNotify)
+        if eg.WindowsVersion >= 'Vista':
```
Comment (topic2k) [inline plugins/System/PowerBroadcastNotifier.py]: needs to be `if eg.WindowsVersion() >= 'Vista':`

**Comment by kdschlosser** [comment]
see #211

**Comment by kdschlosser** [comment]
I made the changes... hopefully this is good to go.

**Pull request merged**